torchvision/prototype/datasets/_builtin/celeba.py

```diff
@@ -128,23 +130,41 @@ def _collate_and_decode_sample(
         path, buffer = image_data
         _, ann = ann_data
 
-        image = decoder(buffer) if decoder else buffer
+        if decoder:
+            image = decoder(buffer)
+            image_size = image.shape[-2:]
+        else:
+            with PIL.Image.open(buffer) as image:
+                image_size = image.height, image.width
+            buffer.seek(0)
```

Comment:
unrelated: why do we need this? Shouldn't this be the job of whatever is the `decoder`?

Reply:
We have a feature branch where the decoding is completely removed from datasets so this is just a crutch for the current state to have working examples. The point here is to get the image size without actually decoding the image.

Reply:
I think the large number of concurrent and experimental PRs make reviewing hard and hinter clarity of direction. It might be worth dedicating some time to clean up (similar to #5377) and merging existing PRs prior sending more.